file-system/fake/file.go

```diff
@@ -0,0 +1,110 @@
+package filesystemfake
+
+import (
+	"bytes"
+	"errors"
+	"os"
+	"time"
+)
+
+var (
+	ErrFileOperationNotPermitted = errors.New("fake file does not exist and therefore this operation is not permitted")
```

Comment:
If we would use `errgo.Newf()` here, we get the codeline where the error is defined too